modules/wallet/wallet.go

```diff
@@ -4,44 +4,51 @@ import (
 	"errors"
 	"log"
 
+	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/sync"
 	"github.com/NebulousLabs/Sia/types"
 )
 
 const (
-	// AgeDelay indicates how long the wallet will wait before allowing the
-	// user to double-spend a transaction under standard circumstances. The
-	// rationale is that most transactions are meant to be submitted to the
-	// blockchain immediately, and ones that take more than AgeDelay blocks
-	// have probably failed in some way.
-	AgeDelay = 80
-
-	// TransactionFee is yet another deprecated-on-arrival constant that says
-	// how large the transaction fees should be. This should really be a
-	// function supplied by the transaction pool.
-	TransactionFee = 10
+	// RespendTimeout records the number of blocks that the wallet will wait
+	// before spending an output that has been spent in the past. If the
+	// transaction spending the output has not made it to the transaction pool
+	// after the limit, the assumption is that it never will.
+	RespendTimeout = 40
 )
 
-type Wallet struct {
-	unlocked bool
-	settings WalletSettings
-
-	state            modules.ConsensusSet
-	tpool            modules.TransactionPool
-	unconfirmedDiffs []modules.SiacoinOutputDiff
-	siafundPool      types.Currency
+var (
+	errLockedWallet    = errors.New("wallet must be unlocked before it can be used")
+	errNilConsensusSet = errors.New("wallet cannot initialize with a nil consensus set")
+	errNilTpool        = errors.New("wallet cannot initialize with a nil transaction pool")
+)
 
-	consensusHeight  types.BlockHeight
-	age              int
-	keys             map[types.UnlockHash]*key
-	timelockedKeys   map[types.BlockHeight][]types.UnlockHash
-	visibleAddresses map[types.UnlockHash]struct{}
-	siafundAddresses map[types.UnlockHash]struct{}
-	siafundOutputs   map[types.SiafundOutputID]types.SiafundOutput
+type spendableKey struct {
+	unlockConditions types.UnlockConditions
+	secretKeys       []crypto.SecretKey
+}
 
-	generatedKeys map[types.UnlockHash]generatedSignatureKey
-	trackedKeys   map[types.UnlockHash]struct{}
+type Wallet struct {
+	unlocked    bool
+	subscribed  bool
+	settings    WalletSettings
+	primarySeed modules.Seed
+
+	state              modules.ConsensusSet
+	tpool              modules.TransactionPool
+	consensusSetHeight types.BlockHeight
+	siafundPool        types.Currency
+
+	keys            map[types.UnlockHash]spendableKey
+	siacoinOutputs  map[types.SiacoinOutputID]types.SiacoinOutput
+	siafundOutputs  map[types.SiafundOutputID]types.SiafundOutput
+	historicOutputs map[types.OutputID]types.Currency
+	spentOutputs    map[types.OutputID]types.BlockHeight
+
+	walletTransactions            []modules.WalletTransaction
+	walletTransactionMap          map[modules.WalletTransactionID]*modules.WalletTransaction
```

Comment:
The slice is in chronological order.

Almost did a linked list instead of a slice.